CRM/Contribute/BAO/Contribution.php

```diff
@@ -3592,6 +3604,28 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
         }
       }
     }
+    if ($context == 'changePaymentInstrument') {
```

Comment:
yeah, agreed. Can you please update the wiki and also inform @JoeMurray about the changes?